### DISCUSSION TITLE
Reusable styles function return type

### DISCUSSION THREAD

**Discussion by richbachman** [wayfair/rainbow-sprinkles#103] [Q&A] [answered]
Hi. I'm trying to create a reusable function that returns a set of styles that can be spread on a component. Here's the code for reference:

```
import React from 'react';
import { Box } from '../../index';
import type {Sprinkles} from '../../styles/atoms.css';

interface ButtonProps {
  children: React.ReactNode;
}

const baseStyles = (): Sprinkles => {
  return {
    all: 'unset',
    padding: '$space10',
  };
};

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  ({ children, ...props }: ButtonProps, ref): JSX.Element => {
    return (
      <Box
        as="button"
        backgroundColor="$colorBackgroundPrimary"
        color="$colorTextInverse"
        ref={ref}
        {...baseStyles()}
        {...props}
      >
        {children}
      </Box>
    );
  }
);

Button.displayName = 'Button';

export { Button };
```

My initial thought was to set the return type to `Sprinkles`, but am running into a TS error:
<img width="839" alt="Screenshot 2023-02-08 at 5 01 16 PM" src="https://user-images.githubusercontent.com/1350081/217671024-941503ca-d2ea-4485-bbb3-9363a407318d.png">

Do you have ideas on what I should be setting that return type to? Thanks in advance for your help!

**Comment by richbachman** [comment]
I found something similar here in this [issue](https://github.com/wayfair/rainbow-sprinkles/issues/42#issuecomment-1176166423), but what is `SystemProp`?

**Reply by roginfarrer** [reply, marked as answer]
You might try the new `satisfies` operator if your version of TS supports it. `as const` may also work. I imagine TS is not inferring the value `'unset'` literally, and instead just taking it as `string`

```tsx
const baseStyles = () => {
  return {
    all: 'unset',
    padding: '$space10',
  } satisfies Sprinkles;
};
```

**Reply by richbachman** [reply]
I forgot about trying `satisfies`. That's a good solution. Thanks for the help!
